source/docs/software/networktables/using-messagepack.rst

```diff
@@ -0,0 +1,124 @@
+                public RawPublisher publisher;
+
+                public Publisher() {
+                    inst.startClient4("radar target publisher");
+                    inst.setServer("localhost", NetworkTableInstance.kDefaultPort4);
+                    publisher = inst.getTable("radar").getRawTopic("targets").publish("msgpack");
+                }
+
+                public void accept(TargetList list) {
+                    publisher.set(objectMapper.writeValueAsBytes(list));
+                }
+            }
+
+On the receiver side, the story is similar, with the same client and table, but
+instead of creating a publisher, it adds a listener to "value" events.  The
+listener extracts the raw bytes from the event, and uses the ObjectMapper to
+deserialize back into a TargetList, which is passed to the downstream consumer
+(in this case, an AWT-based renderer, see the
+`project <https://github.com/truher/radar/>`__ for that).
+
+.. tabs::
+
+    .. group-tab:: Java
+
+        .. code-block:: java
+
+            public class Subscriber {
+                public ObjectMapper objectMapper = new ObjectMapper(new MessagePackFactory());
+                public NetworkTableInstance inst = NetworkTableInstance.getDefault();
+                public Consumer<TargetList> consumer;
+
+                public Subscriber(Consumer<TargetList> consumer) {
+                    this.consumer = consumer;
+                    inst.startClient4("radar target subscriber");
+                    inst.setServer("localhost", NetworkTableInstance.kDefaultPort4);
+                    inst.addListener(
+                        inst.getTable("radar").getEntry("targets"),
+                        EnumSet.of(NetworkTableEvent.Kind.kValueAll),
+                        (event) -> render(event));
+                }
+
+                private void render(NetworkTableEvent event) {
+                    consumer.accept(
+                        objectMapper.readValue(event.valueData.value.getRaw(), TargetList.class));
+                }
+            }
+
+And there you have it!  POJO's passed end to end courtesy of Network Tables 4,
+MessagePack for Java, and Jackson.
+
+Native Dependencies
+-------------------
+
+To use Network Tables, you need to correctly load the native ntcore library,
+which itself depends on two other native wpi libraries, and these need to be
+correctly handled in the build and app itself.  Also, some parts of wpimath
+(e.g. aspects of the pose used in the Target above) involve matrix math from
+another native library,
+`EJML <https://ejml.org/wiki/index.php?title=Main_Page>`__.
+The details of Gradle and native loaders are not complicated, but the error
+messages can be hard to interpret when things go wrong, for example, an error
+about missing dependent libraries doesn't mention **which** dependent library
+is missing.  See `the project <https://github.com/truher/radar>`__
+for details in build.gradle and how to load native code in the application.
```

Comment:
We already have a section on building desktop NT applications, so this section isn't needed (or should be merged there if relevant).

Reply:
yeah, i started by following those instructions, but they didn't work for me.  i struggled with the build for a long time, actually, so i felt like i needed to say something about it; i didn't end up with a gradle file that looked very much like the existing doc.  i'm not sure what the sources of trouble really were.  inclusion of dependent native libs? usage of beta artifacts?  the nonfunctional static initializer loaders?  i'm definitely not an expert with any of the gradle stuff, but i'm sure almost none of the wpilib users are either, so i wanted to provide something that would work unmodified.  i guess this is kinda the same issue as the first one above: i'd like to be able to provide something that will "just work".